rasa/core/tracker_store.py

```diff
@@ -1115,3 +1113,22 @@ def _load_from_module_name_in_endpoint_config(
             f"Using `InMemoryTrackerStore` instead."
         )
         return InMemoryTrackerStore(domain)
+
+
+@contextlib.contextmanager
+def tracker_store_with_full_conversation_retrieval(
+    tracker_store: "TrackerStore",
+) -> Generator[None, None, None]:
+    """Context manager that ensures `tracker_store` fetches all tracker sessions.
+
+    Args:
+        tracker_store: Tracker store to modify.
+    """
+    previous_value = tracker_store.load_events_from_previous_conversation_sessions
+
+    tracker_store.load_events_from_previous_conversation_sessions = True
+
+    try:
```

Comment:
I'm a bit worried about the implementation as generator. Somebody might see and use this function with an `await` expression inside which would have side effects on other requests which are using trackers.

* Why not making it atomic - means returning the `DialogueStateTracker` right away?
* An alternative would be to add a new function `retrieve_full_tracker` to `TrackerStore` and implement this for the `TrackerStore` class and the trackers which are supporting sessions. By default it could just point to `retrieve`

Reply:
I'm not sure I follow - why would someone use this in an `await` expression? It is a context manager - the decorator and the function description says so.

I could also add an explicit example of how to use it in the function docstring. what do you think?

Reply:
I meant the following use case:

```python
        with rasa.core.tracker_store.tracker_store_with_full_conversation_retrieval(
            self.tracker_store
        ):
            tracker = self.tracker_store.get_or_create_tracker(
                conversation_id, append_action_listen=False
            )
            # imagine any `async` call here
            # the await will allow other requests to be processed while this is blocking
            # These requests will then use a tracker store with disabled sessions.
            await asyncio.sleep(123)
```

Reply:
I don't think so - other calls to `self.tracker_store` within that class will happen _outside_ the context manager, so should not be affected by this

Reply:
I agree with the suggestion to make this a tracker store method 👍